Extract media related code from EditPostActivity to a new class

oguzkocer opened the pull request:
This PR is a part of the ongoing refactor for the `EditPostActivity`. Borrowing from the very similar [photo picker PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10608):

> The extraction is almost a one to one conversion where the old implementation is kept the same, sometimes (possibly) to a fault. The goal is moving common logic into new components rather than improving said components, so the review should be done with this in mind. Having said that, some basic improvements are made where it's obvious that the previous logic is untouched.

This extraction is a lot trickier than the photo picker one and considering we are planning to refactor the `EditorMedia` component further, I think it's best to merge this PR into a feature branch instead of `develop`. This branch shouldn't live for a long time, so merge conflicts shouldn't be a big deal. I am also unsure about how best to test these changes, I just don't know enough about the `EditPostActivity`.

_P.S: I have not been able to test this PR yet due to a crash I am experiencing in `develop` and in this branch. I'll mark the PR as draft for now._

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -916,6 +894,7 @@ public void onPhotoPickerHidden() {
     public void onPhotoPickerMediaChosen(@NonNull final List<Uri> uriList) {
         mEditorPhotoPicker.hidePhotoPicker();
 
+        // TODO: It looks like we might be calling addMediaList twice
```
I'm not sure I understand this comment correctly, but haven't you just missed the `return` statement after ```WPMediaUtils.advertiseImageOptimization(this, () -> mEditorMedia.addMediaList(uriList, false));```?

oguzkocer replied:
I don't really remember what I was thinking with this. The `TODO` doesn't make sense 😕 Removed in 48f8d02d516a1ac3b0c3960910961cfe54bf4b9c.

malinajirka commented:
Thanks @oguzkocer!

I've reviewed the code as best I could and it LGTM as a first iteration. Tbh I don't think we need to test this branch before we merge it into the working branch. We'll change big chunks of the code anyway so it feels like a waste of time. 

SideNote: I've already started refactoring `AddMediaListThread` so I run some basic tests there and it worked as expected.

I'm not sure why the tests are timing out 🤷‍♂

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/ui/posts/editor/EditorMedia.kt:
```diff
@@ -0,0 +1,522 @@
+                }
+            }
+
+
```
Fixed in 10e24b799ce05e68c086347a9045c25c60cb107f.

oguzkocer commented:
@malinajirka I fixed the lint issues and removed that `TODO` you left a comment on. I also tested basic functionality. I agree that we shouldn't spend too much time on testing right now. Marking this PR as ready to review!
